pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -70,13 +70,14 @@ protected CompletableFuture<LookupData> internalLookupTopicAsync(final TopicName
                     // Currently, it's hard to check the non-persistent-non-partitioned topic, because it only exists
                     // in the broker, it doesn't have metadata. If the topic is non-persistent and non-partitioned,
                     // we'll return the true flag.
-                    CompletableFuture<Boolean> existFuture = (!topicName.isPersistent() && !topicName.isPartitioned())
-                            ? CompletableFuture.completedFuture(true)
-                            : pulsar().getNamespaceService().checkTopicExists(topicName)
-                                .thenCompose(exists -> exists ? CompletableFuture.completedFuture(true)
-                                        : pulsar().getBrokerService().isAllowAutoTopicCreationAsync(topicName));
-
-                    return existFuture;
+                    return pulsar().getNamespaceService().checkTopicExists(topicName, true).thenCompose(info -> {
+                        boolean exists = info.isExists();
+                        info.recycle();
+                        if (exists) {
+                            return CompletableFuture.completedFuture(true);
+                        }
+                        return pulsar().getBrokerService().isAllowAutoTopicCreationAsync(topicName);
+                    });
```

Comment:
> Instead, we can just return true here if the topic is a non-persistent topic.

After this change, the below three behaviors 

1. Disable auto-creation, the `non-persistent` topic does not exist
  a. response: `{partitions: 0}`, but the expected behavior should be `Topic No Found`. In my opinion, the response `{partitions: 0}` is also okay, because it is the original behavior before the current PR.
2. Enable auto-creation; `allowAutoTopicCreationType=partitioned`; `defaultNumPartitions: 3`. 
  a. response: `{partitions: 0}`, but the expected behavior should be `{partitions: 3}`. In my opinion, `{partitions: 0}` is better than `{partitions: 3}`, because it can avoid auto create partitioned non-persistent topics.

Reply:
Sorry, I misunderstood before, the param `assumedNonPartitionedNonPersistentTopicAlwaysExists` of `NamespaceService.checkTopicExists()` has been removed